atlassian_jwt_auth/contrib/django/middleware.py

```diff
@@ -1,14 +1,25 @@
+import warnings
+
 from django.conf import settings
 
+import atlassian_jwt_auth
+from ..server.helpers import _requires_asap
+from .utils import parse_jwt, verify_issuers, _build_response
+
 
 class ASAPForwardedMiddleware(object):
     """Enable client auth for ASAP-enabled services that are forwarding
     non-ASAP client requests.
 
     This must come before any authentication middleware.
+
+    DEPRECATED: use ASAPMiddleware instead.
```

Comment:
Why are we deprecating this ?

Reply:
Because it relies on client auth succeeding before ASAP, which is backwards.